book/src/proposals/bankless-leader.md

```diff
@@ -0,0 +1,59 @@
+# Bankless Leader
+
+A bankless leader does the minimum amount of work to produce a valid
+block.  The leader is tasked with ingress transactions, sorting and
+filtering valid transactions, arranging them into entries, shredding
+the entries and broadcasting the shreds.
+
+While a validator only needs to reassemble the block and replay
+execution of well formed entries.  For transactions that have
+low execution costs, the leader does 3x more memory operations than
+the validator.
+
+## Execution Key
+
+The [execution key](terminology.md#execution_key) pays for the
+transaction to be included in the block.  The leader only needs to
+validate that the execution key has the balance to pay for the
+fee.
+
+## Balance Cache
+
+For the duration of the block, the leader maintains a temporary
+balance cache for all the processed execution keys.  The cache is
+a map of pubkeys to lamports.
+
+At the start of the block the balance cache is empty.  At the end
+of the block the cache is destroyed.
+
+## Balance Check
+
```

Comment:
What signatures are verified in the Transaction?  Just the execution key's signature or all?

Reply:
all are verified